src/psyclone/psyir/nodes/container.py

```diff
@@ -172,6 +172,7 @@ def __str__(self):
         return f"Container[{self.name}]\n"
 
     def find_routine_psyir(self, name, allow_private=False,
+                           follow_imports=True,
                            check_wildcard_imports=False):
```

Comment:
See my previous comment, not following_imports and not checking_wildcard_imports will potentially return wrong results. Unless this is a necessary assumption for LFRic I suggest making these mandatory (not arguments). If the consequences of this are too big I am happy to rethink this.

Reply:
I'm not quite sure I follow. I've simplified the routine so that it now only ever checks the current container -  it doesn't look at imports, whether wildcard or otherwise. It doesn't affect the statistics of kernels offloaded in GungHo. Is that OK?